### PR TITLE
Add OnnxConfig for SqueezeBert iss17314

### PR DESCRIPTION
**Pull request by artemisep**
# What does this PR do?

## Who can review?
@lewtun @LysandreJik 

Anyone in the community is free to review the PR once the tests have passed.

### CODE REVIEW

**Comment by Ruihua-Fang** [comment]
I run pytest on those 4 failed tests in CircleCI shown above on my local dev, no error message was found, all says skipped except the following one says passed
tests/models/pegasus/test_modeling_pegasus.py::PegasusStandaloneDecoderModelTest::test_sample_generate

File: src/transformers/models/squeezebert/configuration_squeezebert.py [hunk 1]
```diff
@@ -154,3 +157,19 @@ def __init__(
         self.post_attention_groups = post_attention_groups
         self.intermediate_groups = intermediate_groups
         self.output_groups = output_groups
+
+
+class SqueezeBertOnnxConfig(OnnxConfig):
```
Comment (lewtun) [inline src/transformers/models/squeezebert/configuration_squeezebert.py:162]: Could you please add this to ensure we propagate changes in the source model (should they arise):

```suggestion
# # Copied from transformers.models.bert.configuration_bert.BertOnxxConfig with Bert->SqueezeBert
class SqueezeBertOnnxConfig(OnnxConfig):
```

File: src/transformers/onnx/features.py [hunk 2]
```diff
@@ -351,6 +352,16 @@ class FeaturesManager:
             "token-classification",
             onnx_config_cls=RoFormerOnnxConfig,
         ),
+        "squeezebert": supported_features_mapping(
+            "default",
+            "masked-lm",
+            "causal-lm",
```
Comment (lewtun) [inline src/transformers/onnx/features.py:358]: If I'm not mistaken, SqueezeBert doesn't have a causal LM head, so I suggest removing this feature
```suggestion
```

**Review by lewtun — APPROVED** [review]
Thanks for adding this model to the ONNX exports @Ruihua-Fang ! I think those tests were flaky and unrelated to your PR - I've rerun the CI pipeline to see if that's true.

I've left a few minor comments, but this looks good to me. Could you please confirm that the slow tests pass with:

```
RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -s -k "squeezebert"
```

Optionally, feel free to add an exported checkpoint to the ONNX working group org on the Hub: https://huggingface.co/OWG

Gently pinging @sgugger for final approval

**Review by sgugger — APPROVED** [review]
Thanks for your PR! LGTM with Lewis comments.

**Review by LysandreJik — APPROVED** [review]
LGTM, let's merge!

**Pull request merged**